src/KafkaFlow.Abstractions/IProducerContext.cs

```diff
@@ -19,5 +19,12 @@ public interface IProducerContext
         /// Gets the partition offset associated with the message
         /// </summary>
         long? Offset { get; }
+
+        /// <summary>
+        /// Gets an instance of IDependencyResolver which provides methods to resolve dependencies.
+        /// This instance is tied to the producer scope, meaning it is capable of resolving dependencies
+        /// that are scoped to the lifecycle of a single producer.
+        /// </summary>
+        IDependencyResolver ProducerDependencyResolver { get; }
```

Comment:
Do we need the Producer prefix here? 
Asking, since I'm looking at the IMessageContext at the same time.

Reply:
agreed